google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb

```diff
@@ -198,4 +199,22 @@
     file_default_acl.reload!
     _(file_default_acl.acl.readers).must_equal ["allUsers"]
   end
+focus
+  it "sets public_access_prevention to enforced and is unable to modify bucket ACL rules" do
+    refute bucket.public_access_prevention_enforced?
+    bucket.uniform_bucket_level_access = "enforced"
+    assert bucket.public_access_prevention_enforced?
+    err = expect do
+      bucket.acl.public!
+    end.must_raise Google::Cloud::InvalidArgumentError
+  end
+focus
+  it "sets enforced public_access_prevention to nil and is able to modify bucket ACL rules" do
+    refute bucket.public_access_prevention_enforced?
+    bucket.uniform_bucket_level_access = "enforced"
+    assert bucket.public_access_prevention_enforced?
+    bucket.uniform_bucket_level_access = nil
+    refute bucket.public_access_prevention_enforced?
+    bucket.acl.public!
+  end
```

Comment:
@shaffeeullah Do my projects need to be enabled for this feature? I currently see the following expectation failures:

```
  1) Failure:
Google::Cloud::Storage::Bucket::uniform_bucket_level_access::storage#test_0013_sets public_access_prevention to enforced and is unable to modify bucket ACL rules [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb:206]:
Expected false to be truthy.

  2) Failure:
Google::Cloud::Storage::Bucket::uniform_bucket_level_access::storage#test_0014_sets enforced public_access_prevention to nil and is able to modify bucket ACL rules [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb:215]:
Expected false to be truthy.
```

Reply:
@shaffeeullah @tritone I provided our google-cloud-ruby project IDs for the allowlist in a comment on the design doc, thank you.

Reply:
@quartzmo It looks like both of these projects are already on the allowlist. As noted in @tritone's comment in the design doc, I think these tests are still expected to fail, as ACL permissions do not yet work as expected.

Reply:
Thank you. And good news, I got the correct error today!

```
Class: <Google::Cloud::FailedPreconditionError>
Message: <"conditionNotMet: The member bindings allUsers and allAuthenticatedUsers are not allowed since public access prevention is enforced.">
```